src/index.ts

```diff
@@ -181,66 +184,66 @@ function startApp(appSchema, path: string) {
-          accessToken,
-          userID,
-          defaultTimezone,
-          ...loaders,
-          // For stitching purposes
-          exchangeSchema,
-          requestIDs,
-          userAgent,
-        }
+    const graphqlServer = graphqlHTTP((req, res, params) => {
+      const accessToken = req.headers["x-access-token"] as string | undefined
+      const appToken = req.headers["x-xapp-token"] as string | undefined
+      const userID = req.headers["x-user-id"] as string | undefined
+      const timezone = req.headers["x-timezone"] as string | undefined
+      const userAgent = req.headers["user-agent"]
+
+      const { requestIDs = {} } = res.locals ?? {}
+      const requestID = requestIDs.requestID
+
+      if (enableRequestLogging) {
+        fetchLoggerSetup(requestID)
+      }
+
+      // Accepts a tz database timezone string. See http://www.iana.org/time-zones,
+      // https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
+      let defaultTimezone
+      if (timezone && moment.tz.zone(timezone)) {
+        defaultTimezone = timezone
+      }
+
+      const loaders = createLoaders(accessToken, userID, {
+        requestIDs,
+        userAgent,
+        appToken,
+      })
 
-        const validationRules = [principalFieldDirectiveValidation]
-        if (QUERY_DEPTH_LIMIT)
-          validationRules.push(depthLimit(QUERY_DEPTH_LIMIT))
+      const context: ResolverContext = {
+        accessToken,
+        userID,
+        defaultTimezone,
+        ...loaders,
+        // For stitching purposes
+        exchangeSchema,
+        requestIDs,
+        userAgent,
+      }
+
+      const validationRules = [principalFieldDirectiveValidation]
+      if (QUERY_DEPTH_LIMIT) validationRules.push(depthLimit(QUERY_DEPTH_LIMIT))
+
+      return {
+        schema: appSchema,
+        graphiql: !PRODUCTION_ENV,
+        context,
+        rootValue: {},
+        customFormatErrorFn: graphqlErrorHandler(enableSentry, {
+          req,
+          // Why the checking on params? Do we reach this code if params is falsy?
+          variables: params && params.variables,
+          query: (params && params.query)!,
+        }),
+        validationRules,
+        extensions: ({ document, result }) =>
+          createExtensions(document, result, requestID),
+      }
+    })
 
-        return {
-          schema: appSchema,
-          graphiql: !PRODUCTION_ENV,
-          context,
-          rootValue: {},
-          customFormatErrorFn: graphqlErrorHandler(enableSentry, {
-            req,
-            // Why the checking on params? Do we reach this code if params is falsy?
-            variables: params && params.variables,
-            query: (params && params.query)!,
-          }),
-          validationRules,
-          extensions: ({ document, result }) =>
-            createExtensions(document, result, requestID),
-        }
-      })
-    )
+    app.use("/batch", bodyParser.json(), graphqlBatchHTTPWrapper(graphqlServer))
```

Comment:
I'm trying to get a better sense of what this is doing. So it receives requests at `/batch` and delegates those to graphqlServer. Is it `/v2/batch` or just `/batch` at the root? If the latter, does it always get serviced by the v2 schema?

Reply:
both, since these are mounted on both routes.